src/onedrive.d

```diff
@@ -629,8 +630,57 @@ final class OneDriveApi
 		try {
 			http.perform();
 		} catch (CurlException e) {
-			// Potentially Timeout was reached on handle error
-			log.error("ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?");
+			// Parse and display error message received from OneDrive
+			log.error("ERROR: OneDrive returned an error with the following message:");
+			auto errorArray = splitLines(e.msg);
+			string errorMessage = errorArray[0];
+			
+			if (canFind(errorMessage, "Couldn't connect to server on handle")) {
+				// This is a curl timeout
+				log.error("  Error Message: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?");
+				// or 408 request timeout
+				// https://github.com/abraunegg/onedrive/issues/694
+				// Back off & retry with incremental delay
+				int retryCount = 10000;
+				int retryAttempts = 1;
+				int backoffInterval = 2;
+				int maxBackoffInterval = 3600;
+				bool retrySucess = false;
+				while (!retrySucess){
+					log.vdebug("  Retry Attempt: ", retryAttempts);
+					int thisBackOff = retryAttempts*backoffInterval;
+					if (thisBackOff <= maxBackoffInterval) {
+						Thread.sleep(dur!"seconds"(retryAttempts*backoffInterval));
```

Comment:
Could be `Thread.sleep(dur!"seconds"(thisBackOff))`

Reply:
correct